tests/conftest.py

```diff
@@ -159,3 +159,30 @@ def s3_bucket(request: pytest.FixtureRequest):
         return 'my-bucket'
     else:
         return _get_option(request.config, 's3_bucket')
+
+
+# WARNING: These fixtures should be copied if used in a test, as they are session scope to avoid
+# multiple calls to the HuggingFace hub
+@pytest.fixture(scope='session')
+def tiny_bert_model():
+    transformers = pytest.importorskip('transformers')
+
+    config = transformers.AutoConfig.from_pretrained('prajjwal1/bert-tiny')
+    hf_model = transformers.AutoModelForMaskedLM.from_config(config)  # type: ignore (thirdparty)
+    return hf_model
+
+
+@pytest.fixture(scope='session')
+def tiny_bert_tokenizer():
+    transformers = pytest.importorskip('transformers')
+
+    hf_tokenizer = transformers.AutoTokenizer.from_pretrained('prajjwal1/bert-tiny')
+    return hf_tokenizer
+
+
+@pytest.fixture(scope='session')
+def tiny_bert_config():
+    transformers = pytest.importorskip('transformers')
+
+    hf_config = transformers.AutoConfig.from_pretrained('prajjwal1/bert-tiny')
+    return hf_config
```

Comment:
So, my main concern here is someone will forget to deepcopy and other things will break in confusing ways that are hard to debug. I wonder if we can memoize with`functools.lru_cache` again?

Reply:
yeah it is a fair concern. Unfortunately lru cache also returns a single object, so also needs to be deep copied

```
In [3]: @functools.lru_cache
   ...: def list_function():
   ...:     return [1,2,3]
   ...: 

In [4]: a = list_function()

In [5]: b = list_function()

In [6]: a[0] = 5

In [7]: a
Out[7]: [5, 2, 3]

In [8]: b
Out[8]: [5, 2, 3]
```

Reply:
resolving offline